x-pack/platform/plugins/shared/entity_manager/server/lib/v2/definitions/instance_as_filter.test.ts

```diff
@@ -0,0 +1,181 @@
+    const instance: EntityV2 = {
+      'entity.type': 'my_type',
+      'entity.id': 'whatever',
+      'entity.display_name': 'Whatever',
+    };
+
+    const sources: EntitySourceDefinition[] = [
+      {
+        id: 'my_source',
+        type_id: 'my_type',
+        identity_fields: ['host.name'],
+        index_patterns: [],
+        metadata_fields: [],
+        filters: [],
+      },
+    ];
+    readSourceDefinitionsMock.mockResolvedValue(sources);
+
+    expect(instanceAsFilter(instance, esClientMock, logger)).rejects.toThrowError(
+      InvalidEntityInstance
+    );
+  });
+
+  it('creates a single source filter for a single identity field', () => {
+    const instance: EntityV2 = {
+      'entity.type': 'my_type',
+      'entity.id': 'whatever',
+      'entity.display_name': 'Whatever',
+      'host.name': 'my_host',
+    };
+
+    const sources: EntitySourceDefinition[] = [
+      {
+        id: 'my_source',
+        type_id: 'my_type',
+        identity_fields: ['host.name'],
+        index_patterns: [],
+        metadata_fields: [],
+        filters: [],
+      },
+    ];
+    readSourceDefinitionsMock.mockResolvedValue(sources);
+
+    expect(instanceAsFilter(instance, esClientMock, logger)).resolves.toBe('(host.name: my_host)');
+  });
+
+  it('creates a single source filter for multiple identity field', () => {
+    const instance: EntityV2 = {
+      'entity.type': 'my_type',
+      'entity.id': 'whatever',
+      'entity.display_name': 'Whatever',
+      'host.name': 'my_host',
+      'host.os': 'my_os',
+    };
+
+    const sources: EntitySourceDefinition[] = [
+      {
+        id: 'my_source',
+        type_id: 'my_type',
+        identity_fields: ['host.name', 'host.os'],
+        index_patterns: [],
+        metadata_fields: [],
+        filters: [],
+      },
+    ];
+    readSourceDefinitionsMock.mockResolvedValue(sources);
+
+    expect(instanceAsFilter(instance, esClientMock, logger)).resolves.toBe(
+      '(host.name: my_host AND host.os: my_os)'
+    );
+  });
+
+  it('creates multiple source filters for a single identity field', () => {
+    const instance: EntityV2 = {
+      'entity.type': 'my_type',
+      'entity.id': 'whatever',
+      'entity.display_name': 'Whatever',
+      'host.name': 'my_host',
+      'host.os': 'my_os',
+    };
+
+    const sources: EntitySourceDefinition[] = [
+      {
+        id: 'my_source_host',
+        type_id: 'my_type',
+        identity_fields: ['host.name'],
+        index_patterns: [],
+        metadata_fields: [],
+        filters: [],
+      },
+      {
+        id: 'my_source_os',
+        type_id: 'my_type',
+        identity_fields: ['host.os'],
+        index_patterns: [],
+        metadata_fields: [],
+        filters: [],
+      },
+    ];
+    readSourceDefinitionsMock.mockResolvedValue(sources);
+
+    expect(instanceAsFilter(instance, esClientMock, logger)).resolves.toBe(
+      '(host.name: my_host) OR (host.os: my_os)'
```

Comment:
Q: can a situation like this actually happen? An entity definition with inconsistent `identity_fields`?

Reply:
Yes, if you have a type called `host`, and then you attach two sources.

In source 1 the `identity_fields` is set to `host.name` (based on the data in the target indices for that source) and in source 2 it's set to `host.hostname` for example because the indices for source 2 are strictly ECS so they don't have `host.name`. 

So it can easily happen.